### ISSUE TITLE
Odd quirk with arrow keys and mouse location

### ISSUE THREAD

**Issue by ladlon** [turtletooth/tahoma2d#459]
I noticed that if the mouse pointer is over the timeline, the left/right arrows move the selection box left/right on the timeline (along 'time'), and the up/down arrow moves it up/down (through the various columns).  So, that's all good.

However, if the mouse pointer is over the Viewer window, it appears that BOTH the left/right AND up/down arrows move the timeline selection box left/right along the timeline (rather than the up/down arrows moving it up/down, through the various columns, as it does if the mouse is over the timeline).  Not sure why, and if it's intentional, I'm not sure what the point of having a double mapping of the left/right movement (especially at the cost of the up/down functionality being lost).

Does that happen with the default key mappings?  I have a custom set, so I'm not sure if that's a factor.

Tahoma2D 1.0.1
Win7Pro

**Comment by manongjohn** [comment]
That is normal behavior for viewer and the arrow keys.  As for why the double mapping? I can't say, though I suppose it's easier to tap up/down keys with 1 finger than it is with left/right.  May have to do with the fact that OT originally only had an Xsheet, so perhaps the up/down was kept with that in mind.

**Comment by ladlon** [comment]
Ya, I don't get the logic... The only difference between the two scenarios is where the mouse pointer is located at the time.

The arrow key mapping makes sense in the timeline (up is up, etc)... but if the mouse is over the viewer (as it often will be as you move items around in the viewer to animate it), up is.... left... and left is left.... so you can't move between columns/items without moving the mouse pointer over the timeline (which kind of defeats the purpose of using the keys, since you almost might as well just click the column at that point).  I really don't get why the up/down is redundantly remapped to left/right.

As always, I suggest letting the user set it whatever why works best for them.

As it stands, I now have to constantly move the mouse pointer up and down (from timeline, to viewer, to timeline....) just to keep the functionality I set up with the shortcuts.

Hopefully, that gets fixed...

**Comment by ladlon** [comment]
Tahoma2D overall has an odd thing with keyboard shortcuts not maintaining their functionality from window to window.  The disabling of the timeline transport shortcuts in the Graph Editor is baffling and frustrating.

**Comment by manongjohn** [comment]
When the mouse is over the viewer, the viewer has focus and since it only has a time element, it can only move from frame to frame.

Personally I don't think when you are over the viewer that the arrow keys should move columns. I am likely in the minority there.

I would not want to accidentally move columns and start working, only to find out I spent the last 30 minutes on the wrong column.  And what does one do when someone uses both an xsheet and timeline in the same room (some people may do that) or switches from one to the other in the middle of the session? I would find that rather confusing.

**Comment by ladlon** [comment]
Valid points, which is why I always say, let the user configure it as they see fit... that way, everyone wins.

**Comment by ladlon** [comment]
In my case, for example, I am animating two eyebrows individually... so, I'm going to a point in time, moving one eyebrow (or setting a 'pinning' key, selecting the other eyebrow (column), moving it (or setting a key), and so on... one after the other... So, the ability to quickly switch between columns (eyebrows) without having to constantly move the mouse to/from the viewer would be a really good time/effort saver.

**Comment by beeheemooth** [comment]
@manongjohn I agree that the arrow functions should be different depending on where the cursor is (over the viewer, over the timeline or over the palette window) and this is logical. But there are many commands that we cannot access while in another window (or if this window is closed at all). For example, when a user draws in fullscreen mode, he does not have access to some of the options from the Tool Option Bar, he also cannot duplicate the current frame or insert an empty frame, since this requires selecting a cell. If the timeline window is closed (or not in focus), then moving columns or frames using shortcuts "blindly" is a rather strange thing, I agree. However, if in such a situation the user wants to make a duplicate frame or insert a blank frame (or use Drawing Substitution), then it is obvious that he wants to do this in the CURRENT frame in the CURRENT column, and not in the randomly highlighted 508th frame. Switching selection tools (Selection Tool, Control Point Editor) automatically remove the highlight from the frame, and the user can highlight the current frame in the current column only with the cursor.
